hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/handlers/FileGetInfoHandler.java

```diff
@@ -101,7 +102,14 @@ public void validate(@NonNull final QueryContext context) throws PreCheckExcepti
             } catch (IOException e) {
                 throw new RuntimeException("Unable to read file contents", e);
             }
-            optionalInfo.ifPresent(responseBuilder::fileInfo);
+
+            if (optionalInfo.isEmpty()) {
+                responseBuilder.header(header.copyBuilder()
+                        .nodeTransactionPrecheckCode(INVALID_FILE_ID)
+                        .build());
+            } else {
+                optionalInfo.ifPresent(responseBuilder::fileInfo);
```

Comment:
Instead of lambda syntax (which is costly), we've already verified that the optionalInfo is present and not empty, we can just call responseBuilder.fileInfo(optionalInfo.get()); directly.

Reply:
done thanks